src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type InitValue = string;
+
+export function useLocalStorage(
+  key: string,
+  initialValue: InitValue,
+) {
+  const [storedValue, setStoredValue] = useState(() => {
+    if (typeof window === 'undefined') {
+      return JSON.parse(initialValue);
+    }
+
+    try {
+      const item = window.localStorage.getItem(key);
+
+      return item ? JSON.parse(item) : JSON.parse(initialValue);
+    } catch (error) {
+      return JSON.parse(initialValue);
+    }
+  });
+
+  const setValue = (value: Todo[]) => {
+    try {
+      const valueToStore
+        = value instanceof Function ? value(storedValue) : value;
+
+      setStoredValue(valueToStore);
+
+      if (typeof window !== 'undefined') {
+        window.localStorage.setItem(key, JSON.stringify(valueToStore));
+      }
+    } catch (error) {
+      // A more advanced implementation would handle the error case
```

Comment:
Remove comment. If you catch a bug and do nothing else, you're just hiding it. Either handle the error or don't catch it at all, let it pop up above the code